src/os/bluestore/ZonedAllocator.cc

```diff
@@ -235,10 +233,16 @@ void ZonedAllocator::find_zones_to_clean(void) {
 		 << *idx.begin() << " num_dead_bytes = " << zone_states[*idx.begin()].num_dead_bytes
 		 << dendl;
 
-  zones_to_clean = {idx.begin(), idx.begin() + num_zones_to_clean};
+  zones_to_clean = {idx.begin(), idx.begin() + num_zones_to_clean_at_once};
+  num_zones_to_clean = num_zones_to_clean_at_once;
 
   // TODO: handle the case of disk being full.
   ceph_assert(!zones_to_clean.empty());
+  ceph_assert(num_zones_to_clean == 0);
```

Comment:
This should be num_zones_to_clear != 0 and in fact makes a little sense for now given the manipulation on num_zones_to_clean above.

Reply:
Oops. Fixed.